pkg/apis/serving/v1alpha1/service_types.go

```diff
@@ -278,69 +224,22 @@ func (ss *ServiceStatus) PropagateRouteStatus(rs RouteStatus) {
 	}
 	switch {
 	case rc.Status == corev1.ConditionUnknown:
-		ss.markUnknown(ServiceConditionRoutesReady, rc.Reason, rc.Message)
+		condSet.Using(ss).MarkUnknown(ServiceConditionRoutesReady, rc.Reason, rc.Message)
 	case rc.Status == corev1.ConditionTrue:
-		ss.markTrue(ServiceConditionRoutesReady)
+		condSet.Using(ss).MarkTrue(ServiceConditionRoutesReady)
 	case rc.Status == corev1.ConditionFalse:
-		ss.markFalse(ServiceConditionRoutesReady, rc.Reason, rc.Message)
+		condSet.Using(ss).MarkFalse(ServiceConditionRoutesReady, rc.Reason, rc.Message)
 	}
 }
 
-func (ss *ServiceStatus) markTrue(t ServiceConditionType) {
-	ss.setCondition(&ServiceCondition{
-		Type:   t,
-		Status: corev1.ConditionTrue,
-	})
-	for _, cond := range []ServiceConditionType{
-		ServiceConditionConfigurationsReady,
-		ServiceConditionRoutesReady,
-	} {
-		c := ss.GetCondition(cond)
-		if c == nil || c.Status != corev1.ConditionTrue {
-			return
-		}
-	}
-	ss.setCondition(&ServiceCondition{
-		Type:   ServiceConditionReady,
-		Status: corev1.ConditionTrue,
-	})
+// GetConditions returns the Conditions array. This enables generic handling of
+// conditions by implementing the sapis.Conditions interface.
+func (ss *ServiceStatus) GetConditions() []sapis.Condition {
+	return ss.Conditions
 }
 
-func (ss *ServiceStatus) markUnknown(t ServiceConditionType, reason, message string) {
-	ss.setCondition(&ServiceCondition{
-		Type:    t,
-		Status:  corev1.ConditionUnknown,
-		Reason:  reason,
-		Message: message,
-	})
-	for _, cond := range []ServiceConditionType{
-		ServiceConditionConfigurationsReady,
-		ServiceConditionRoutesReady,
-	} {
-		c := ss.GetCondition(cond)
-		if c == nil || c.Status == corev1.ConditionFalse {
-			// Failed conditions trump unknown conditions
-			return
-		}
-	}
-	ss.setCondition(&ServiceCondition{
-		Type:    ServiceConditionReady,
-		Status:  corev1.ConditionUnknown,
-		Reason:  reason,
-		Message: message,
-	})
-}
-
-func (ss *ServiceStatus) markFalse(t ServiceConditionType, reason, message string) {
-	for _, cond := range []ServiceConditionType{
-		t,
-		ServiceConditionReady,
-	} {
-		ss.setCondition(&ServiceCondition{
-			Type:    cond,
-			Status:  corev1.ConditionFalse,
-			Reason:  reason,
-			Message: message,
-		})
-	}
+// SetConditions sets the Conditions array. This enables generic handling of
+// conditions by implementing the sapis.Conditions interface.
```

Comment:
Can you add an assertion that the interface is met with the others?

Reply:
done